File: frontend/src/modules/graph/domain/commit.ts
export interface Commit {
  id: string;
  parents: string[];
  refs: string[];
  authorName: string;
  authorEmail: string;
  authoredAt: string;
  message: string;
  isWorkingChanges?: boolean;
}

export interface GraphCommit extends Commit {
  shortId: string;
  lane: number;
  parentLanes: number[];
  convergingLanes: number[];
  passthroughLanes: number[];
  isBranchTip: boolean;
  laneCount: number;
}

File: backend/src/domain/commit.rs
#[derive(Debug, Clone, PartialEq, Eq)]
pub struct Commit {
    pub id: String,
    pub parents: Vec<String>,
    pub refs: Vec<String>,
    pub author_name: String,
    pub author_email: String,
    pub authored_at: String,
    pub message: String,
}
